src/plots/plots.js

```diff
@@ -287,17 +287,26 @@ plots.resize = function(gd) {
         if(gd._redrawTimer) clearTimeout(gd._redrawTimer);
 
         gd._redrawTimer = setTimeout(function() {
-            if((gd._fullLayout || {}).autosize) {
-                // autosizing doesn't count as a change that needs saving
-                var oldchanged = gd.changed;
+            // return if there is nothing to resize
+            if(gd.layout.width && gd.layout.height) {
+                resolve(gd);
+                return;
+            }
+
+            delete gd._fullLayout._initialAutoSizeIsDone;
+            if(!gd.layout.width) delete (gd._fullLayout || {}).width;
+            if(!gd.layout.height) delete (gd._fullLayout || {}).height;
 
-                // nor should it be included in the undo queue
-                gd.autoplay = true;
+            // autosizing doesn't count as a change that needs saving
+            var oldchanged = gd.changed;
 
-                Plotly.relayout(gd, { autosize: true });
+            // nor should it be included in the undo queue
+            gd.autoplay = true;
+
+            Plotly.plot(gd).then(function() {
```

Comment:
Both `Plotly.plot(gd)` and `Plotly.relayout(gd, { autosize: true })` would have the same effect in terms of resizing, but `Plotly.relayout(gd, { autosize: true })` would have the side-effect of setting `layout.autosize` to true. But:
- if `layout.autosize` was `true`, there is no need to set it to `true` again;
- and if `layout.autosize` was `false`, both `layout.width` and `layout.height` would've already been initialised and setting `layout.autosize` to `true` would have no use.